internal/controller/streams.go

```diff
@@ -328,25 +328,49 @@ func (sm *streamManager) StreamResources(ctx context.Context, logger *zap.Logger
 	}
 	logger.Info("Successfully sent resource snapshot")
 
+	mutationChan := make(chan *pb.KubernetesResourceMutation)
+
 	// PHASE 3: Start watchers concurrently
 	for _, info := range allWatchInfos {
 		resourceManager := resourceManagers[info.resource]
 
 		go func(info watcherInfo, manager *ResourceManager) {
-			manager.WatchK8sResources(ctx, cancel, info.apiGroup, info.resourceVersion)
+			manager.WatchK8sResources(ctx, cancel, info.apiGroup, info.resourceVersion, mutationChan)
 		}(info, resourceManager)
 	}
 
 	dd.mutex.Lock()
 	dd.processingResources = false
 	dd.mutex.Unlock()
 
-	<-ctx.Done()
-	return ctx.Err()
+	ticker := time.NewTicker(jitterTime(keepalivePeriod, 0.10))
```

Comment:
I still think that using a Timer would be more appropriate as it fires only once, whereas a Ticker writes multiple ticker timestamps into the channel in case we missed reading one, e.g., in case a Send takes too long.
And all those ticker timestamps get sent as multiple messages all at once, instead of just one.

Not a blocker, though.